### PR TITLE
Feature/edit and delete category

### PR DESCRIPTION
**Pull request by vinhlee95**

### CODE REVIEW

File: src/pages/Settings/components/CategorySettings.tsx [hunk 1]
```diff
@@ -67,6 +80,18 @@ const CategorySettings: React.FunctionComponent<Props> = ({
 		}
 	}, [category.status])
 
+	const handleDeleteCategory = id => {
+		deleteCategory(team._id, id)
+	}
+
+	const handleUpdateCategoryItem = (categoryItem: Category) => {
+		updateCategory(team._id, categoryItem._id, {
+			name: categoryItem.name,
```
Comment (HuyAms) [inline src/pages/Settings/components/CategorySettings.tsx:89]: could use destructuring

**Review by HuyAms — CHANGES_REQUESTED** [review]

File: src/pages/Settings/components/CategorySettings.tsx [hunk 2]
```diff
@@ -82,6 +107,18 @@ const CategorySettings: React.FunctionComponent<Props> = ({
 			dataIndex: 'description',
 			editable: true,
 		},
+		{
+			title: '',
+			dataIndex: 'operation',
+			render: (text, record: Category) => (
+				<Popconfirm
+					title={t('categories.deleteConfirmation')}
+					onConfirm={() => handleDeleteCategory(record._id)}
```
Comment (HuyAms) [inline src/pages/Settings/components/CategorySettings.tsx:116]: if you have face up with Lambdash are forbidden in JSX. You could walk around by declaring handleDeleteCategory in render function
```

	{
			title: '',
			dataIndex: 'operation',
			render: (text, record: Category) => (
                                 const handleDeleteCategory = () => deleteCategory(team._id, record._id)
				<Popconfirm
					title={t('categories.deleteConfirmation')}
					onConfirm={ handleDeleteCategory}
				>
					<Icon type="delete" theme="twoTone" twoToneColor="red" />
				</Popconfirm>
			),
		},
```

**Review by HuyAms — CHANGES_REQUESTED** [review]

File: src/pages/Settings/components/CategorySettings.tsx [hunk 3]
```diff
@@ -112,7 +149,11 @@ const CategorySettings: React.FunctionComponent<Props> = ({
 
 	const renderCategoryTable = (data: Category[], type: CategoryType) => (
 		<CategoryTableWrapper>
-			<Table columns={getTableColumns(type)} data={data} />
+			<Table
+				columns={getTableColumns(type)}
```
Comment (HuyAms) [inline src/pages/Settings/components/CategorySettings.tsx:153]: loading={status === 'fetch'}

File: src/pages/Settings/components/CategorySettings.tsx [hunk 4]
```diff
@@ -82,6 +107,18 @@ const CategorySettings: React.FunctionComponent<Props> = ({
 			dataIndex: 'description',
 			editable: true,
 		},
+		{
+			title: '',
+			dataIndex: 'operation',
+			render: (text, record: Category) => (
+				<Popconfirm
```
Comment (HuyAms) [inline src/pages/Settings/components/CategorySettings.tsx:114]: (status === 'saving') ? <Spinner/> . : <Popconfirm..../>

This one is a little bit tricky since you have to check the index to render correct spinner position, so let's do this in another PR

**Review by HuyAms — CHANGES_REQUESTED** [review]

File: src/pages/Settings/components/CategorySettings.tsx (hunk 2 above)
Comment (vinhlee95) [reply src/pages/Settings/components/CategorySettings.tsx:116]: I tried this but that lambda error is still there

File: src/pages/Settings/components/CategorySettings.tsx (hunk 4 above)
Comment (vinhlee95) [reply src/pages/Settings/components/CategorySettings.tsx:114]: Yeah I think we can left it later since we are using relatively fast network

**Review by HuyAms — APPROVED** [review]
